src/Saas.Identity/Saas.Permissions/Saas.Permissions.Service/Services/PermissionsService.cs

```diff
@@ -96,12 +96,18 @@ public async Task RemoveUserPermissionsFromTenantAsync(string tenantId, string u
 
     }
 
-    // filter not currently implemented.
+    /// <summary>
+    /// Optionally supply a filter for a specific TenantId
+    /// </summary>
+    /// <param name="userId"></param>
+    /// <param name="filter"></param>
+    /// <returns></returns>
     public async Task<ICollection<string>> GetTenantsForUserAsync(string userId, string? filter)
     {
         _logger.LogDebug("{userId} has requested tenants", userId);
         return await _context.Permissions
             .Where(x => x.UserId == userId)
+            .Where(x => filter == null || x.TenantId.Length == filter.Length && EF.Functions.Like(x.TenantId, $"%{filter}%"))
```

Comment:
Remove this. The filter was leftover from testing and we don't have a use for it in the frontend at the moment.